lib/phobos/actions/process_batch_inline.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'phobos/batch_message'
+require 'phobos/processor'
+
+module Phobos
+  module Actions
+    class ProcessBatchInline
+      include Phobos::Processor
+
+      attr_reader :metadata
+
+      def initialize(listener:, batch:, metadata:)
+        @listener = listener
+        @batch = batch
+        @listener = listener
+        @batch = batch
+        @metadata = metadata.merge(
+          batch_size: batch.messages.count,
+          partition: batch.partition,
+          offset_lag: batch.offset_lag,
+          retry_count: 0
+        )
+      end
+
+      def execute
+        payloads = @batch.messages.map do |message|
+          Phobos::BatchMessage.new(
+            key: message.key,
+            partition: message.partition,
+            offset: message.offset,
+            payload: force_encoding(message.value)
+          )
+        end
+
+        begin
+          process_batch(payloads)
+        rescue StandardError => e
+          handle_error(e, 'listener.retry_handler_error_batch',
+                       "error processing inline batch, waiting #{backoff_interval}s")
+          retry
+        end
+      end
+
+      private
+
+      def process_batch(payloads)
+        instrument('listener.process_batch_inline', @metadata) do |_metadata|
+          handler = @listener.handler_class.new
+
+          preprocessed_payloads = before_consume(handler, payloads)
```

Comment:
I think we should call `preprocessed_payloads = handler.before_consume_batch(payloads, @metadata)` directly, and skip the roundtrip to the function below (which can be deleted).

Reply:
Yep, I'm fine with that - was just mirroring the implementation in `process_message`.